crates/duty-tracker/src/contract_manager.rs

```diff
@@ -58,14 +68,21 @@ impl ContractManager {
             let mut block_sub = zmq_client.subscribe_blocks().await;
             loop {
                 tokio::select! {
-                    Some(block) = block_sub.next() => if let Err(e) = ctx.process_block(block).await {
-                        todo!() // TODO(proofofkeags): error handling
+                    Some(block) = block_sub.next() => {
+                        let blockhash = block.block_hash();
+                        if let Err(e) = ctx.process_block(block).await {
+                            tracing::error!("failed to process block {}: {}", blockhash, e);
+                            break;
+                        }
                     },
                     Some(event) = p2p_handle.next() => match event {
-                        Ok(Event::ReceivedMessage(msg)) => if let Err(e) = ctx.process_p2p_message(msg).await {
-                            todo!() // TODO(proofofkeags): error handling
+                        Ok(Event::ReceivedMessage(msg)) => if let Err(e) = ctx.process_p2p_message(msg.clone()).await {
+                            tracing::error!("failed to process p2p msg {:?}: {}", msg, e);
+                            break;
                         },
-                        Err(e) => todo!(),
+                        Err(e) => {
+                            tracing::error!("{}", e);
```

Comment:
full import
Also we prefer
```suggestion
                            error!(%e "other unknown error");
```